lib/mongo/server/connection_common.rb

```diff
@@ -59,14 +59,16 @@ def connected?
       #   authentication document, if any.
       # @param [ true | false ] load_balancer Whether the connection is to
       #   a load balancer.
+      # @param server_api [ Hash ] server_api Server API version.
       #
       # @return [BSON::Document] Document that should be sent to a server
       #     for handshake purposes.
       #
       # @api private
-      def handshake_document(app_metadata, speculative_auth_doc: nil, load_balancer: false)
-        document = if app_metadata.server_api && app_metadata.server_api[:version]
-                     HELLO_DOC
+      def handshake_document(app_metadata, speculative_auth_doc: nil, load_balancer: false, server_api: {})
+        serv_api = app_metadata.server_api || server_api || {}
+        document = if serv_api[:version]
```

Comment:
I don't think this is equivalent to what was in place before. An api version with, for example, only `:strict` set was previously sent to the server as `apiStrict` parameter but now wouldn't be sent, if I read this correctly.

Reply:
You're right, I didn't notice that.
However, I think that the new behavior is actually correct. According to versioned-api spec server api client options must include `version`, and may include `strict` and `deprecationErrors`. So, we should not send two latter to the server without a version.

Reply:
That is correct - a serverApi object must always contain a version. Only passing a strict or deprecation option is invalid.